be/src/olap/tablet.cpp

```diff
@@ -283,35 +164,70 @@ OLAPStatus Tablet::add_rowset(RowsetSharedPtr rowset, bool need_persist) {
             LOG(FATAL) << "failed to save rowset to local meta store" << rowset->rowset_id();
         }
     }
+    RETURN_NOT_OK(save_meta());
     ++_newly_created_rowset_num;
     return OLAP_SUCCESS;
 }
 
-OLAPStatus Tablet::modify_rowsets(const vector<RowsetSharedPtr>& to_add,
-                                  const vector<RowsetSharedPtr>& to_delete) {
+OLAPStatus Tablet::add_rowsets(const vector<RowsetSharedPtr>& to_add) {
+    WriteLock wrlock(&_meta_lock);
+
+    // Add rowsets to tablet_meta for compaction/clone/alter table/rollup.
+    // Before taking these rowsets into effect, should
+    // check shortest-path existence after adding rowsets
+
     vector<RowsetMetaSharedPtr> rs_metas_to_add;
     for (auto& rs : to_add) {
         rs_metas_to_add.push_back(rs->rowset_meta());
     }
 
+    TabletMetaSharedPtr new_tablet_meta(new TabletMeta());
+    RETURN_NOT_OK(TabletMetaManager::get_meta(_data_dir, tablet_id(), schema_hash(), new_tablet_meta));
+    new_tablet_meta->add_rs_metas(rs_metas_to_add);
```

Comment:
do not use tablet manager get_meta method.

Reply:
OK